app/src/main/java/net/gsantner/markor/format/zimwiki/ZimWikiTextConverter.java

```diff
@@ -0,0 +1,215 @@
+/*#######################################################
+ *
+ *   Maintained by Gregor Santner, 2018-
+ *   https://gsantner.net/
+ *
+ *   License of this file: Apache 2.0 (Commercial upon request)
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+#########################################################*/
+package net.gsantner.markor.format.zimwiki;
+
+import android.arch.core.util.Function;
+import android.content.Context;
+
+import net.gsantner.markor.format.TextConverter;
+import net.gsantner.markor.format.markdown.MarkdownTextConverter;
+import net.gsantner.opoc.util.StringUtils;
+
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Wrapper class around MarkdownTextConverter
+ */
+@SuppressWarnings("WeakerAccess")
+public class ZimWikiTextConverter extends TextConverter {
+
+    private Context _context;
+    private File _file;
+    private String _currentLine;
+
+    private final MarkdownTextConverter _markdownConverter;
+
+    private static final Pattern LIST_ORDERED_LETTERS = Pattern.compile("^\t*([\\d]+\\.|[a-zA-Z]+\\.) ");
+
+    public ZimWikiTextConverter(MarkdownTextConverter markdownConverter) {
+        _markdownConverter = markdownConverter;
+    }
+
+    /**
+     * First, convert zim-wiki to regular Markor markdown. Then, calls the regular converter.
+     *
+     * @param markup              Markup text
+     * @param context             Android Context
+     * @param isExportInLightMode True if the light theme is to apply.
+     * @param file                The file to convert.
+     * @return HTML text
+     */
+    @Override
+    public String convertMarkup(String markup, Context context, boolean isExportInLightMode, File file) {
+        _context = context;
+        _file = file;
+
+        String contentWithoutHeader = markup.replaceFirst(ZimWikiHighlighterPattern.ZIMHEADER.pattern.toString(), "");
+        StringBuilder markdownContent = new StringBuilder();
+
+        for (String line : contentWithoutHeader.split("\\r\\n|\\r|\\n")) {
+            String markdownEquivalentLine = getMarkdownEquivalentLine(line, isExportInLightMode);
+            markdownContent.append(markdownEquivalentLine);
+            markdownContent.append("  "); // line breaks must be made explicit in markdown by two spaces
+            markdownContent.append(String.format("%n"));
+        }
+
+        return _markdownConverter.convertMarkup(markdownContent.toString(), context, isExportInLightMode, file);
+    }
+
+    private String getMarkdownEquivalentLine(String zimWikiLine, final boolean isExportInLightMode) {
+        _currentLine = zimWikiLine;
```

Comment:
Is it required to cache the line here? Should be un-needed duplication that blows up memory size (depending on amount of text)

Reply:
Yes, introducing the field variable makes it easier to manipulate the content of the line in the other methods.
Actually, it does not duplicate the content of the string and blow up memory (except for an additional pointer) because Strings are passed by reference in Java like any other objects, not by value like primitive types - so the two variables refer to the same content in memory.

Reply:
I handled this.